### ISSUE TITLE
Borg readability and coding style

### ISSUE THREAD

**Issue by maltefiala** [borgbackup/borg#4]
I am not really fond of the readability of attic's code. Variable names like "t0" and "st" are nice to write but don't make it very readable and brake PEP 0008 as they clearly aren't words:
> lowercase with words separated by underscores as necessary to improve readability.

However, changing variable names should be discussed as it brakes code compatibility with attic, see #1

**Comment by anarcat** [comment]
i don't think code compatibility with attic is such a big issue as on-disk format compatibility.

**Comment by ThomasWaldmann** [comment]
I added configuration so we are able to run flake8 code style checker locally.
The current configuration is made so it does not trigger any warnings/errors for stuff that is already present in existing code. Later (NOT NOW), when all the merges are done, we can improve style by reducing the excludes in setup.cfg flake8 section and at the same time fixing the style.

**Comment by ThomasWaldmann** [comment]
some stuff reported by flake8 was fixed here (I first checked that it is just minor stuff, not likely to cause merge issues):
https://github.com/borgbackup/borg/commit/4b81f380f882515ecd7a683f371eb0666564a0f8

**Comment by anarcat** [comment]
for future reference, the configuration is done in `setup.cfg`:

```
[flake8]
ignore = E226,F403
max-line-length = 250
exclude = docs/conf.py,borg/_version.py,build,dist,.git,.idea,.cache,.tox
max-complexity = 100
```

F403 is: "`from module import *` used; unable to detect undefined names"
F226 is: "missing whitespace around arithmetic operator"

i believe @ThomasWaldmann was saying that the line limit is 120 characters (and not 250). max complexity, i have no idea what it's for.

even with the above checks skipped, we have 62 messages from flake8:

```
./setup.py:9:1: F401 'DistutilsOptionError' imported but unused
./setup.py:25:17: E225 missing whitespace around operator
./setup.py:77:9: E125 continuation line with same indent as next logical line
./setup.py:126:1: E302 expected 2 blank lines, found 1
./setup.py:132:5: E301 expected 1 blank line, found 0
./setup.py:183:5: E301 expected 1 blank line, found 0
./setup.py:246:5: E122 continuation line missing indentation or outdented
./setup.py:247:5: E122 continuation line missing indentation or outdented
./setup.py:248:5: E122 continuation line missing indentation or outdented
./setup.py:249:5: E122 continuation line missing indentation or outdented
./setup.py:250:1: E122 continuation line missing indentation or outdented
./borg/__init__.py:3:1: F401 '__version__' imported but unused
./borg/__main__.py:3:1: W391 blank line at end of file
./borg/archive.py:567:30: E901 SyntaxError: invalid syntax
./borg/archiver.py:2:32: E111 indentation is not a multiple of four
./borg/archiver.py:2:32: E113 unexpected indentation
./borg/archiver.py:195:13: E129 visually indented line with same indent as next logical line
./borg/archiver.py:289:55: E271 multiple spaces after keyword
./borg/archiver.py:322:119: E901 SyntaxError: invalid syntax
./borg/cache.py:97:29: E901 SyntaxError: invalid syntax
./borg/cache.py:141:15: E271 multiple spaces after keyword
./borg/helpers.py:198:25: E901 SyntaxError: invalid syntax
./borg/key.py:200:60: E901 SyntaxError: invalid syntax
./borg/locking.py:99:25: E221 multiple spaces before operator
./borg/logger.py:36:1: F401 'sys' imported but unused
./borg/remote.py:137:67: E202 whitespace before ']'
./borg/repository.py:283:33: E901 SyntaxError: invalid syntax
./borg/upgrader.py:73:23: E901 SyntaxError: invalid syntax
./borg/xattr.py:53:27: E901 SyntaxError: invalid syntax
./borg/support/__init__.py:16:1: W391 blank line at end of file
./borg/support/argparse.py:97:5: E301 expected 1 blank line, found 0
./borg/support/argparse.py:115:1: E302 expected 2 blank lines, found 1
./borg/support/argparse.py:624:44: E127 continuation line over-indented for visual indent
./borg/support/argparse.py:677:15: E271 multiple spaces after keyword
./borg/support/argparse.py:1159:1: E302 expected 2 blank lines, found 1
./borg/support/argparse.py:1260:5: E303 too many blank lines (2)
./borg/support/argparse.py:1468:57: F812 list comprehension redefines 'action' from line 1463
./borg/support/argparse.py:2092:17: E128 continuation line under-indented for visual indent
./borg/support/argparse.py:2092:21: F812 list comprehension redefines 'action' from line 2082
./borg/support/argparse.py:2092:29: F812 list comprehension redefines 'option_string' from line 2080
./borg/support/argparse.py:2092:44: F812 list comprehension redefines 'explicit_arg' from line 2080
./borg/testsuite/archiver.py:74:1: E302 expected 2 blank lines, found 1
./borg/testsuite/archiver.py:74:29: E901 SyntaxError: invalid syntax
./borg/testsuite/archiver.py:120:5: E301 expected 1 blank line, found 0
./borg/testsuite/archiver.py:238:19: E222 multiple spaces after operator
./borg/testsuite/archiver.py:257:46: E261 at least two spaces before inline comment
./borg/testsuite/benchmark.py:13:1: F401 'cmd' imported but unused
./borg/testsuite/benchmark.py:27:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:53:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:60:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:65:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:70:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:76:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:81:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:86:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:91:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:97:1: F811 redefinition of unused 'cmd' from line 13
./borg/testsuite/benchmark.py:100:1: W391 blank line at end of file
./borg/testsuite/compress.py:102:1: W391 blank line at end of file
./borg/testsuite/locking.py:5:58: E203 whitespace before ','
./borg/testsuite/locking.py:11:1: E302 expected 2 blank lines, found 1
./borg/testsuite/upgrader.py:25:18: F821 undefined name 'Repository'
1       E111 indentation is not a multiple of four
1       E113 unexpected indentation
5       E122 continuation line missing indentation or outdented
1       E125 continuation line with same indent as next logical line
1       E127 continuation line over-indented for visual indent
1       E128 continuation line under-indented for visual indent
1       E129 visually indented line with same indent as next logical line
1       E202 whitespace before ']'
1       E203 whitespace before ','
1       E221 multiple spaces before operator
1       E222 multiple spaces after operator
1       E225 missing whitespace around operator
1       E261 at least two spaces before inline comment
3       E271 multiple spaces after keyword
4       E301 expected 1 blank line, found 0
5       E302 expected 2 blank lines, found 1
1       E303 too many blank lines (2)
9       E901 SyntaxError: invalid syntax
4       F401 'DistutilsOptionError' imported but unused
10      F811 redefinition of unused 'cmd' from line 13
4       F812 list comprehension redefines 'action' from line 1463
1       F821 undefined name 'Repository'
4       W391 blank line at end of file
62
```

i think we should remove the ignores or complexity limitations, because they don't change the error count (maybe they add *two*). lowering the line length to 120 chars adds another 62 warnings.

it seems to me this should be intergrated in our CI workflow, otherwise we'll always be catching up to this. a bit like codecov, but without the emails at each PR. :P

**Comment by anarcat** [comment]
we should also write up a blurb about coding style in development.rst, esp. for those who don't know (or forgot, like me) about flake8.

**Comment by anarcat** [comment]
config fixed and docs done in #318, next step is to do the actual cleanup - not much to be done actually, and hook into CI/github (not sure how).

**Comment by anarcat** [comment]
another todo here, from #3, is to take a look at the codeclimate stuff:

* https://codeclimate.com/github/jborg/attic/code
* https://codeclimate.com/github/borgbackup/borg

(merged #3 here)

**Comment by ThomasWaldmann** [comment]
add automated pep8/flakes test after 1.0.
